CMakeLists.txt

```diff
@@ -142,47 +141,54 @@ if(BUILD_BINDINGS)
 
   add_external_package(pybind11 VERSION 2.3.0 CONFIG)
 
-   # Get where python site-packages is
-  if(NOT USER)
-    execute_process ( COMMAND "${PYTHON_EXECUTABLE}" -c "from distutils.sysconfig import \
-                      get_python_lib; print(get_python_lib())"
-      OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE
-      )
-  else()
-    # Get where python local site-packages is
-    execute_process ( COMMAND "${PYTHON_EXECUTABLE}" -c "import site; \
-                      print(site.USER_SITE)"
-      OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE
-      )
-  endif()
+
+  execute_process ( COMMAND "${PYTHON_EXECUTABLE}" -c "from distutils.sysconfig import \
+                    get_python_lib; print(get_python_lib())"
+    OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
 
   execute_process ( COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; \
                     print(sys.prefix)"
     OUTPUT_VARIABLE PACKAGE_PREFIX OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
+  )
 endif()
 
 # figure out where to install the library
 if(NOT "${INSTALL_PATH}" STREQUAL "")
   set(CMAKE_INSTALL_PREFIX "${INSTALL_PATH}")
-elseif(NOT BUILD_BINDINGS AND NOT USER)
+elseif(NOT BUILD_BINDINGS)
   set(CMAKE_INSTALL_PREFIX "/usr")
   message(STATUS "INSTALLATION REQUIRES SUDO RIGHTS")
-elseif(NOT BUILD_BINDINGS AND USER)
-  set(CMAKE_INSTALL_PREFIX "$ENV{HOME}/.local")
-elseif(BUILD_BINDINGS AND NOT USER)
+elseif(BUILD_BINDINGS)
   set(CMAKE_INSTALL_PREFIX "${PACKAGE_PREFIX}")
-elseif(BUILD_BINDINGS AND USER)
-  set(CMAKE_INSTALL_PREFIX "$ENV{HOME}/.local")
 endif()
 
 message(STATUS "Installation ROOT: ${CMAKE_INSTALL_PREFIX}")
 
-# Always use full RPATH
+if(SKBUILD)
+  # make sure we know when installing with setup.py (potentially for using with
+  # pip)
+  message(STATUS "The project is built using scikit-build")
+endif()
+
+# relative RPATH so that _rascal.so can find librascal.so when installed.
+# see link for more details on the topic
 # https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
+if(NOT APPLE AND UNIX)
+  # this link path accommodates most installations
+  set(CMAKE_INSTALL_RPATH "\\$ORIGIN/../../../../")
```

Comment:
I don't like this. It is assuming that python packages are always installed at `/usr/local/lib/pythonX.Y/site-packages/`, which is not true. For example, I sometimes use homebrew on linux to install software, in particular modern Python, which then install packages to a different location.

What is wrong with `set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")`?

I think the proper solution here would be to remove the separation between librascal.so and _rascal.cpython-XXX.so, but as a middle ground how about always put `librascal.so` in `site-packages/rascal-egg/lib/librascal.so`, and use the second RPATH only?

Reply:
> I don't like this. It is assuming that python packages are always installed at /usr/local/lib/pythonX.Y/site-packages/, which is not true. For example, I sometimes use homebrew on linux to install software, in particular modern Python, which then install packages to a different location.

I am not sure to follow. You mean to say that one might install python packages not in `somewhere/lib/pythonX.Y/site-packages/` or `~/.local/lib/pythonX.Y/site-packages/` ? Is this actually following the general gidelines of python's package managing ?

> What is wrong with set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")?

if you ran the command `python setup.py install` you can see that the package is not installed from the build package but installed locally and then copied over twice. It seems that it is one of the PEPXXX requirements. So `CMAKE_INSTALL_PREFIX` is not proper.

> I think the proper solution here would be to remove the separation between librascal.so and _rascal.cpython-XXX.so, but as a middle ground how about always put librascal.so in site-packages/rascal-egg/lib/librascal.so, and use the second RPATH only?

having `librascal.so` in the lib folder of the python installation was decided upon a while ago as the **proper** way to do it. Note that it is how it is done with many packages that rely on dynamic c/c++ libraries.
Installing with pip (meaning `wheel`) or with `python setup.py install` (meaning `egg`) does not lead to the same behaviour. In on case you have this egg folder that contains lib while in the other librascal.so is in the main lib folder. I think that both should be handled separately as I could not find a way to block using `egg`. For sure one **should not** use `egg` for installation as this package is not registered within pip.

Reply:
> I am not sure to follow. You mean to say that one might install python packages not in somewhere/lib/pythonX.Y/site-packages/ or ~/.local/lib/pythonX.Y/site-packages/ ? Is this actually following the general gidelines of python's package managing ?

Sorry, what I mean is that assuming the existence of a file outside of `$PREFIX/lib/pythonX.Y/site-packages/rascal/` may backfire, and indeed the file only exist if the whole package was installed using pip.

I also somehow misunderstood what `/../../../../` meant in the RPATH, could you add a comment that this is to find a lib in `$PREFIX/lib` when the bindings are installed in `$PREFIX/lib/pythonX.Y/site-packages/rascal/`?

Thinking a bit more about it, it should indeed work in most settings, so I would be happy with a comment explaining it. I would still prefer not polluting `$PREFIX/lib` when installing a Python package though.

> having librascal.so in the lib folder of the python installation was decided upon a while ago as the proper way to do it. Note that it is how it is done with many packages that rely on dynamic c/c++ libraries.

Do you mean having `$PREFIX/lib/librascal.so` and `$PREFIX/lib/pythonX.Y/site-packages/rascal/lib/_rascal.cpython.X.Y.so`? If so I agree this is the right way to do it when one want to install both the C++ lib and the Python package. But here we are talking about user interested in installing the python lib only, and the C++ lib is only an implementation detail.

> For sure one should not use egg for installation as this package is not registered within pip.

I fail to see the link between using egg and registering with PyPI.

I use `python setup.py install`  and `python setup.py develop` often for Python packages, so I expect other would do the same, hence why I think we should make it work.

Reply:
> Do you mean having $PREFIX/lib/librascal.so and $PREFIX/lib/pythonX.Y/site-packages/rascal/lib/_rascal.cpython.X.Y.so? If so I agree this is the right way to do it when one want to install both the C++ lib and the Python package. But here we are talking about user interested in installing the python lib only, and the C++ lib is only an implementation detail.

the end goal here is to potentially have librascal.so to be used by other executable, e.g. lammps for instance, so that the when building a model with python and running it in c++ uses the same underlying library avoiding incompatibilities and such silly things. So ultimately it would not be only the python lib that we are interested in. 
Of course we can discuss this but that would be outside of this PR.

> I fail to see the link between using egg and registering with PyPI.

PyPI is different from pip since the later is a local command line utility for package managing while the former is an online python package repository that can be accessed with pip. 
What I mean is that a package installed using `python setup.py install` is not managed by any package manager while `pip install .` results in such a package. Hence on should prefer `pip install .` for simplicity of management. Also egg does not follow any PEP standard probably resulting in the difference in behaviour that we observe and the workarounds needed to make it work for our purpose.

> I use python setup.py install and python setup.py develop often for Python packages, so I expect other would do the same, hence why I think we should make it work.

and it does work now, it just needed some additional workaround.